client/app/scripts/superdesk/ui/ui.js

```diff
@@ -1222,7 +1222,47 @@
         };
     }
 
-    angular.module('superdesk.ui', ['superdesk.dashboard.world-clock'])
+    /*
+     * Media Query directive is used for creating responsive
+     * layout's for single elements on page
+     *
+     * Usage:
+     * <div sd-media-query min-width='650' max-width='1440'></div>
+     *
+     */
+    mediaQuery.$inject = ['$window', '$timeout'];
+    function mediaQuery($window, $timeout) {
+        return {
+            scope: {
+                minWidth: '=',
+                maxWidth: '='
+            },
+            link: function (scope, elem) {
+                var window = angular.element($window);
+
+                window.on('resize', _.debounce(calcSize, 300));
+
+                function calcSize() {
+                    $timeout(function () {

```

Comment:
i guess you don't need to call `$timeout` which would trigger full digest, probably you should call only 
```
scope.$parent.$applyAsync(function() { scope.$parent.elementState = 'compact'; });
```